apps/OboeTester/app/src/main/cpp/OboeTesterStreamCallback.cpp

```diff
@@ -38,3 +42,41 @@ void OboeTesterStreamCallback::printScheduler() {
     }
 #endif
 }
+
+// Sleep to cause an XRun. Then reschedule.
+void OboeTesterStreamCallback::maybeHang(const int64_t startNanos) {
+    if (mHangTimeMillis == 0) return;
+
+    if (startNanos > mNextTimeToHang) {
+        LOGD("%s() start sleeping", __func__);
+        // Take short naps until it is time to wake up.
+        int64_t nowNanos = startNanos;
+        int64_t wakeupNanos = startNanos + (mHangTimeMillis * 1000L * 1000L);
+        while (nowNanos < wakeupNanos && mHangTimeMillis > 0) {
+            int32_t sleepTimeMicros = (int32_t) ((wakeupNanos - nowNanos) / 1000);
+            // Don't usleep for more than a second.
```

Comment:
I'm a bit confused by these comments given that the min setting is 4ms and the max is 5 seconds

Reply:
I improved the comments.